modules/seqexec/server/src/main/scala/seqexec/server/gems/GemsControllerEpics.scala

```diff
@@ -3,156 +3,154 @@
+          epicsSys.waitForStableLoops(LoopStabilizationTimeout)
+      }
     }
-  }
 
-  import Gems.Cwfs1DetectorState.cwfs1DetectorStateDetectorStateOps
-  import Gems.Cwfs2DetectorState.cwfs2DetectorStateDetectorStateOps
-  import Gems.Cwfs3DetectorState.cwfs3DetectorStateDetectorStateOps
-
-  private def stopCwfs(current: EpicsGems, demand: GemsConfig): (EpicsGems, Option[F[Unit]]) = {
-    val p = List(
-      (DetectorStateOps.isActive(current.cwfs1) && !demand.isCwfs1Used).option(
-        (EpicsGems.cwfs1.set(Cwfs1DetectorState.Off), epicsSys.ApdControl.setApd1Cmd(ApdStop))
-      ),
-      (DetectorStateOps.isActive(current.cwfs2) && !demand.isCwfs2Used).option(
-        (EpicsGems.cwfs2.set(Cwfs2DetectorState.Off), epicsSys.ApdControl.setApd2Cmd(ApdStop))
-      ),
-      (DetectorStateOps.isActive(current.cwfs3) && !demand.isCwfs3Used).option(
-        (EpicsGems.cwfs3.set(Cwfs3DetectorState.Off), epicsSys.ApdControl.setApd3Cmd(ApdStop))
+    import Gems.Cwfs1DetectorState.cwfs1DetectorStateDetectorStateOps
+    import Gems.Cwfs2DetectorState.cwfs2DetectorStateDetectorStateOps
+    import Gems.Cwfs3DetectorState.cwfs3DetectorStateDetectorStateOps
+
+    private def stopCwfs(current: EpicsGems, demand: GemsConfig): (EpicsGems, Option[F[Unit]]) = {
+      val p = List(
+        (DetectorStateOps.isActive(current.cwfs1) && !demand.isCwfs1Used).option(
+          (EpicsGems.cwfs1.set(Cwfs1DetectorState.Off), epicsSys.ApdControl.setApd1Cmd(ApdStop))
+        ),
+        (DetectorStateOps.isActive(current.cwfs2) && !demand.isCwfs2Used).option(
+          (EpicsGems.cwfs2.set(Cwfs2DetectorState.Off), epicsSys.ApdControl.setApd2Cmd(ApdStop))
+        ),
+        (DetectorStateOps.isActive(current.cwfs3) && !demand.isCwfs3Used).option(
+          (EpicsGems.cwfs3.set(Cwfs3DetectorState.Off), epicsSys.ApdControl.setApd3Cmd(ApdStop))
+        )
+      ).flattenOption
+
+      (
+        p.map(_._1).foldLeft(current){case (a, b) => b(a)},
+        p.nonEmpty.option{
+          p.traverse(_._2) *>
+          epicsSys.ApdControl.setTimeout(CmdTimeout) *>
+          epicsSys.ApdControl.post.void
+        }
       )
-    ).flattenOption
+    }
+
+    private def startCwfs(current: EpicsGems, demand: GemsConfig): Option[F[Unit]] = {
+      val p = List(
+        (!DetectorStateOps.isActive(current.cwfs1) && demand.isCwfs1Used).option(
+          epicsSys.ApdControl.setApd1Cmd(ApdStart)
+        ),
+        (!DetectorStateOps.isActive(current.cwfs2) && demand.isCwfs2Used).option(
+          epicsSys.ApdControl.setApd2Cmd(ApdStart)
+        ),
+        (!DetectorStateOps.isActive(current.cwfs3) && demand.isCwfs3Used).option(
+          epicsSys.ApdControl.setApd3Cmd(ApdStart)
+        )
+      ).flattenOption
 
-    (
-      p.map(_._1).foldLeft(current){case (a, b) => b(a)},
       p.nonEmpty.option{
-        p.traverse(_._2) *>
-        epicsSys.ApdControl.setTimeout(CmdTimeout) *>
-        epicsSys.ApdControl.post.void
+        p.sequence *>
```

Comment:
What is the difference between `p.sequence` and `p.traverse(_._2)`?d

Reply:
This wasn't really changed, It looks like a change because indentation changed making for a large diff